### ISSUE TITLE
About the evaluation of U-IDS/P-IDS

### ISSUE THREAD

**Issue by Haoru** [fenglinglwb/MAT#86]
I give paths of my results and GTs to evaluation/cal_fid_pids_uids.py, and get [pids: 0.0000, uids: 0.0000].  I don't know what caused this. Could you provide some suggestions?

**Comment by fenglinglwb** [comment]
The results of pids and uids may be 0s when the quality isn't good enough. You may verify this by using better results.

**Comment by Haoru** [comment]
I use the better results, and the fid is better, while the pids and uids are still 0s.

**Comment by fenglinglwb** [comment]
It's weird. Maybe you could use our CelebA results to test whether the evaluation script runs normally.

**Comment by Haoru** [comment]
I use the results of MAT_CelebA-HQ_512_SmallMask and get [fid: 3.9669, pids: 0.0408, uids: 0.1871]. Is this within the realm of normalcy?

**Comment by fenglinglwb** [comment]
The results should be 2.86 21.15, and 32.56. Please check whether the ground truth correctly matches the testing result.

**Comment by Haoru** [comment]
I found the corresponding ground truth with the same name as the testing result from the Celeba-512 folder. Is this a correct match?

**Comment by chencc5956** [comment]
I also meet the same problem, and I set the test results and ground truth both the same name in two different file folder, but the p-ids and u-ids are 0.

**Comment by fenglinglwb** [comment]
P-IDS and U-IDS are basically classification scores. If the provided results are not good enough, the score could be 0

**Comment by chencc5956** [comment]
However, I tried to use a script to test the two folders for identical ground truth images. The results were [fid: -0.0001, pids: 0.0000, uids: 0.5000]. Is this correct? Thank you for your answer.

**Comment by fenglinglwb** [comment]
It seems there is something wrong. How could FID be negative?

**Comment by chencc5956** [comment]
If two identical sets of images are used for testing, theoretically the FID should be 0 because they are identical. However, due to numerical calculation errors, a negative number very close to 0 may be obtained, such as -0.0001.  But I don't know why the values of pids and uids are still so small.

**Comment by fenglinglwb** [comment]
If two identical sets are provided, how could they be distinguishable?

**Comment by chencc5956** [comment]
I understand，thank you！

**Comment by chencc5956** [comment]
I have another question, if the metric used in the training process is PSNR or FID, will the models of the two ultimately be significantly different?

**Comment by fenglinglwb** [comment]
We claim that pixel-level PSNR is not a proper metric for large-hole inpainting as it is an ill-posed problem. Multiple possible solutions correspond to a single input. Thus, FID/P-IDS/U-IDS these perceptual metrics could be more reasonable.

**Comment by chencc5956** [comment]
I also agree with this. But the metric I used in training before was psnr. If using psnr or fid in training has a significant impact on the final model generated by training, then I may need to try using fid and retrain again. So will different metrics during the training process result in significant differences in the final model?
